### PR TITLE
[TT-10085] Remove schema details from oas request validation

### PR DESCRIPTION
**Pull request by furkansenharputlu**
Ticket: https://tyktech.atlassian.net/browse/TT-10085

### CODE REVIEW

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5765/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6823164975)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5765/merge
Commit: 46bb4a93e885304e5d6d7027c198965120003dca 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6963087660)

File: gateway/mw_oas_validate_request.go [hunk 1]
```diff
@@ -61,6 +63,8 @@ func (k *ValidateRequest) ProcessRequest(w http.ResponseWriter, r *http.Request,
 		errResponseCode = validateRequest.ErrorResponseCode
 	}
 
+	openapi3.SchemaErrorDetailsDisabled = validateRequest.SchemaErrorDetailsDisabled
```
Comment (jeffy-mathew) [inline gateway/mw_oas_validate_request.go:66]: this is a global variable in `openapi3` isn't there a chance that this could affect other apis as well?

Comment (furkansenharputlu) [reply gateway/mw_oas_validate_request.go:66]: As it is set in every api call before validation, each api will set its value to that and it won’t affect other APIs, right?

Regarding API creation, this is request validation. I couldn’t make sure if the object validation used the same flag for that. Let’s check.

Comment (jeffy-mathew) [reply gateway/mw_oas_validate_request.go:66]: > As it is set in every api call before validation, each api will set its value to that and it won’t affect other APIs, right?
`openapi3.SchemaErrorDetailsDisabled` is not reset back to previous value. even if we do, concurrent requests are not protected right?

Comment (furkansenharputlu) [reply gateway/mw_oas_validate_request.go:66]: Yes, there may be concurrency issue here. I didn't want to add a mutex directly. The library also doesn't provide it to us. It seems we should a mutex right? Or any other solution suggestion?

Comment (jeffy-mathew) [reply gateway/mw_oas_validate_request.go:66]: Another option is to make this a gateway level config, thereby the behaviour stays consistent across all APIs + control APIs

File: apidef/oas/operation.go [hunk 2]
```diff
@@ -517,6 +517,8 @@ type ValidateRequest struct {
 	// ErrorResponseCode is the error code emitted when the request fails validation.
 	// If unset or zero, the response will returned with http status 422 Unprocessable Entity.
 	ErrorResponseCode int `bson:"errorResponseCode,omitempty" json:"errorResponseCode,omitempty"`
+	// SchemaErrorDetailsDisabled disables exposing of details about schema errors.
```
Comment (andyo-tyk) [inline apidef/oas/operation.go:520]:
```suggestion
	// SchemaErrorDetailsDisabled configures how much detail is provided in the error message if the request fails schema validation:
	// If set to `true`, only the relevant part of the schema will be provided
	// If set to `false`, the full expected schema will be provided in the error message
```

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Suggested improved wording for the GoDocs (based on my understanding of the topic). Happy to alter the wording if this isn't quite correct.

**Review by titpetric — APPROVED** [review]

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5765/merge
Commit: 0c8e9132b0457de484c7c2123b65ef7559eb4149 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7004792361)

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5765/merge
Commit: 36710c9930fb6745668d6d858c58ceb9e3a02f36 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7004800643)

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5765/merge
Commit: 36710c9930fb6745668d6d858c58ceb9e3a02f36 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7004800643)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5765/merge
Commit: 36710c9930fb6745668d6d858c58ceb9e3a02f36 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7004800583)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5765/merge
Commit: 0c8e9132b0457de484c7c2123b65ef7559eb4149 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7004792430)

**Review by andyo-tyk — APPROVED** [review]
The change that I was interested in has been removed, so I'm happy to approve based on approvals from others.

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5765/merge
Commit: e652097a0ece6dbbdca6e0a8b04bdbb65874ae41 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7004880954)

**Pull request merged**